File: ui_examples/views/HTMLTextEx01/app/src/main/java/cf/domone/android/html_text_ex_01/MainActivity.java
package cf.domone.android.html_text_ex_01;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Html;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Define a text with HTML tags.
        String htmlText = "This is <font color=#0000FF>blue</font> and <i>italic</i>.";

        TextView tvHtmlTextExample = (TextView)findViewById(R.id.tvHtmlTextExample);
        tvHtmlTextExample.setText(Html.fromHtml(htmlText));
    }
}

File: ui_examples/views/ActivityResultEx/ex01/src/main/java/lee/dorian/android/activity_result_ex01/MainActivity.kt
package lee.dorian.android.activity_result_ex01

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.widget.AppCompatButton

class MainActivity : AppCompatActivity() {

    private lateinit var btnToJoinFormActivity: AppCompatButton
    private lateinit var tvWelcome: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnToJoinFormActivity = findViewById(R.id.btnToJoinFormActivity)
        tvWelcome = findViewById(R.id.tvWelcome)

        btnToJoinFormActivity.visibility = View.VISIBLE
        tvWelcome.visibility = View.GONE

        btnToJoinFormActivity.setOnClickListener(btnToJoinFormActivityClickListener)
    }

    private val btnToJoinFormActivityClickListener = View.OnClickListener {
        val intent = Intent(this, JoinFormActivity::class.java)
        joinFormActivityResultLauncher.launch(intent)
    }

    private val joinFormActivityResultLauncher: ActivityResultLauncher<Intent> = registerForActivityResult(
        ActivityResultContracts.StartActivityForResult()
    ) {
        if (it.resultCode != RESULT_OK) {
            return@registerForActivityResult
        }

        btnToJoinFormActivity.visibility = View.GONE
        tvWelcome.apply {
            val name = it.data?.getStringExtra(JoinFormActivity.IntentExtraKey.NICKNAME) ?: "anonymous"
            text = "Welcome, ${name}."
            visibility = View.VISIBLE
        }
    }

}